hydra-node/src/Hydra/HeadLogic/Outcome.hs

```diff
@@ -164,6 +234,9 @@ wait reason = Wait reason []
 newState :: StateChanged tx -> Outcome tx
 newState change = Continue [change] []
 
+newStateWithEffect :: StateChanged tx -> Outcome tx
+newStateWithEffect change = Continue [change] [ClientEffect change]
```

Comment:
If ServerOutputs are being decoupled from HeadLogic, perhaps we could introduce an EventSink that generates API outputs on demand.

Not sure if this is within the scope of this PR, but we might consider running a separate thread to handle projecting server outputs instead of embedding this loop in the node logic.

WDYT?

Reply:
API should become just an `EventSink` for these effects, look at this one https://github.com/cardano-scaling/hydra/issues/1853 
I think playing with threads is out of scope of this PR since it was supposed to be a simple thing that leads to changes we want here https://github.com/cardano-scaling/hydra/issues/1618